vassal-app/src/main/java/VASSAL/script/ExpressionInterpreter.java

```diff
@@ -275,12 +276,22 @@ else if (BeanShell.TRUE.equals(value)) {
         else if (BeanShell.FALSE.equals(value)) {
           setVar(var, false);
         }
+        // Do not attempt to convert int or float Numeric Literals, return them as Strings.
+        else if (StringUtils.containsAny(value, "dDfFlL")) { // NON-NLS
```

Comment:
Shouldn't it check more specifically if any of those are at the end?

Reply:
Technically yes, but if those characters appear anywhere except at the end, the parseFloat will fail anyway.
Have a look at this version. If the string contains anything except +-.[0-9] we want to short-circuit the parseInt() and parseFloat() and just return it as a String.